fix: propagate exception to fail task if we fail to install the CLI

maxjeffos opened the pull request:
This change catches and then re-throws an exception relating to not being able to install the Snyk CLI. This way it will cause a job to fail when the Snyk CLI cannot be installed.

pavel-snyk commented on src/main/java/io/snyk/jenkins/tools/SnykInstaller.java:
```diff
@@ -121,12 +121,12 @@ private FilePath installSnykAsNpmPackage(FilePath expected, Node node, TaskListe
   }
 
-  private FilePath installSnykAsSingleBinary(FilePath expected, Node node, TaskListener log) throws IOException, InterruptedException {
+  private FilePath installSnykAsSingleBinary(FilePath expected, Node node, TaskListener log) throws IOException, InterruptedException, ToolDetectionException {
```
@maxjeffos, I think there is no need to add ToolDetectionException in method signature, because it extends IOException and we already have it.

maxjeffos replied:
Ok, makes sense - will remove it.
Thanks @pavel-snyk

pavel-snyk commented:
@maxjeffos, thanks for the PR, LGTM (only one small comment)

maxjeffos commented:
I added

> Change itself looks good - but could we do more for whoever runs into that issue? Known causes, what could they try, docs link?

I added an piece on this to the README.md
